src/Microsoft.ML.Featurizers/DateTimeTransformer.cs

```diff
@@ -0,0 +1,834 @@
+        internal struct TimePoint
+        {
+            public int Year;
+            public byte Month;
+            public byte Day;
+            public byte Hour;
+            public byte Minute;
+            public byte Second;
+            public byte AmPm;
+            public byte Hour12;
+            public byte DayOfWeek;
+            public byte DayOfQuarter;
+            public ushort DayOfYear;
+            public ushort WeekOfMonth;
+            public byte QuarterOfYear;
+            public byte HalfOfYear;
+            public byte WeekIso;
+            public int YearIso;
+            public string MonthLabel;
+            public string AmPmLabel;
+            public string DayOfWeekLabel;
+            public string HolidayName;
+            public byte IsPaidTimeOff;
+
+            internal unsafe TimePoint(byte* rawData)
+            {
+                int intPtrSize = sizeof(IntPtr);
+
+                Year = *(int*)rawData;
+                rawData += 4;
+
+                Month = *rawData++;
+                Day = *rawData++;
+                Hour = *rawData++;
+                Minute = *rawData++;
+                Second = *rawData++;
+                AmPm = *rawData++;
+                Hour12 = *rawData++;
+                DayOfWeek = *rawData++;
+                DayOfQuarter = *rawData++;
+                DayOfYear = *(ushort*)rawData;
+                rawData += 2;
+
+                WeekOfMonth = *(ushort*)rawData;
+                rawData += 2;
+
+                QuarterOfYear = *rawData++;
+                HalfOfYear = *rawData++;
+                WeekIso = *rawData++;
+                YearIso = *(int*)rawData;
+                rawData += 4;
+
+                // Convert char * to string
+                MonthLabel = GetStringFromPointer(ref rawData, intPtrSize);
+                AmPmLabel = GetStringFromPointer(ref rawData, intPtrSize);
+                DayOfWeekLabel = GetStringFromPointer(ref rawData, intPtrSize);
+                HolidayName = GetStringFromPointer(ref rawData, intPtrSize);
+                IsPaidTimeOff = *rawData;
+            }
+
+            // Converts a pointer to a native char* to a string and increments pointer by to the next value.
+            // The length of the string is stored at byte* + sizeof(IntPtr).
+            private static unsafe string GetStringFromPointer(ref byte* rawData, int intPtrSize)
+            {
+                byte[] buffer;
+                if (intPtrSize == 4) // 32 bit machine
+                    buffer = new byte[*(uint*)(rawData + intPtrSize)];
+                else // 64 bit machine
+                    buffer = new byte[*(ulong*)(rawData + intPtrSize)];
+
+                if (buffer.Length == 0)
+                {
+                    rawData += intPtrSize * 2;
+                    return string.Empty;
+                }
+
+                Marshal.Copy(new IntPtr(*(int**)rawData), buffer, 0, buffer.Length);
```

Comment:
If you are using .NET Core 2.1 and above, you can save this buffer and copy and use `Encoding.UTF8.GetString(ReadOnlySpan<byte>)`, where the Span points directly to native memory. It is one less alloc and copy.

Reply:
How can I tell if its using .NET Core 2.1 or above? I thought I was, but I dont have the method signature you listed. I only have the one that takes a `byte[]`. All I am seeing is that the target framework is `netstandard2.0`.

Reply:
You need to cross-target the project for multiple frameworks. Today the project only builds for `netstandard2.0`. 

https://github.com/dotnet/machinelearning/blob/9fc8f1cd323d572672a5460914f927acfef310dc/src/Microsoft.ML.Featurizers/Microsoft.ML.Featurizers.csproj#L4

But if you change that to

```xml
<TargetFrameworks>netstandard2.0;netcoreapp2.1</TargetFrameworks>
```

Then the project builds 2 dlls - one for netstandard 2.0 and one for netcoreapp2.1.

And then in your code you can use a `#if` to detect which one you are building for:

```C#
#if NETSTANDARD2_0
// current code
#else
// use Encoding.UTF8.GetString(ReadOnlySpan<byte>)
#endif
```